### ISSUE TITLE
How can I create a dataset similar to KITTI?

### ISSUE THREAD

**Issue by mobimg** [jbehley/point_labeler#95]
Dear Professor Behley,

I hope this message finds you well. I am currently working on creating a dataset in the KITTI format using your point_labeler tool. The data is collected via a LiDAR-equipped drone, stored in ROS bag files. Could you kindly provide a brief overview of the production process for the KITTI dataset, including the methods employed?

I apologize for any inconvenience this request may cause and deeply appreciate your guidance. Thank you for developing such a valuable tool and sharing your expertise through your datasets.

Best regards.

**Comment by jbehley** [comment]
As I didn't produce the KITTI dataset itself, I can only say what is needed to potentially label the point clouds with the tool (which was built for working with KITTI format files.) There are potentially other ways, but it's your decision to use it or not.

1. Pose estimation: The point_labeler uses the poses provided in the `poses.txt`. These can be estimated with something like KISS-ICP or KISS-SLAM that provide an easy-to-use off-the-shelf solution that is easy to setup: `pip install` is all you need. It usually provide good poses out of the box without extensive hyperparameter tuning. For drone data, it might be even worth considering a LiDAR-intertial SLAM or odmetry approach (e.g. [3,4]) as the "motion profile" is not as "constant" as on a car driving around.
2. Calibration: If you have only a single sensor and the poses are anyway in the coordinate system of the sensor, you have to simply have a identity matrix in the calibration file. (You find some issues here that discuss this)
3. The point_labeler currently works just with the point clouds in KITTI format; therefore you would have to produce the `.bin` files (or write something to directly ready the rosbags by modifying the code)
4. Motion compensation aka point cloud deskewing: In KITTI, the point clouds are already provided in way that accounts for the motion of the sensor while scanning the environment; i.e., the movement between the start of a turn of the spinning LIDAR is already accounted for. KISS-ICP does a point cloud deskewing internally to estimate the poses, but it would be good to also produce the point clouds for labeling such that points are deskewed.

However, all the steps above require probably some fine-tuning with the specific platform that you are using, as I expect that the drone movement is more "high frequent" than a "normal" car drive; therefore, integrating information of an IMU is beneficial to account for the motion of the sensor while scanning the environment (it's similar to the rolling shutter vs. a global shutter with images).


References
[1] https://github.com/PRBonn/kiss-icp
[2] https://github.com/PRBonn/kiss-slam
[3] https://github.com/YibinWu/LIO-EKF
[4] https://github.com/hku-mars/FAST_LIO

**Comment by mobimg** [comment]
thank you for your help. It will helpful for me&nbsp;




最美的太阳
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jbehley/point_labeler"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月26日(星期三) 晚上9:37
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jbehley/point_labeler] How can I create a dataset similar to KITTI? (Issue #95)



   
As I didn't produce the KITTI dataset itself, I can only say what is needed to potentially label the point clouds with the tool (which was built for working with KITTI format files.) There are potentially other ways, but it's your decision to use it or not.
  
Pose estimation: The point_labeler uses the poses provided in the poses.txt. These can be estimated with something like KISS-ICP or KISS-SLAM that provide an easy-to-use off-the-shelf solution that is easy to setup: pip install is all you need. It usually provide good poses out of the box without extensive hyperparameter tuning. For drone data, it might be even worth considering a LiDAR-intertial SLAM or odmetry approach (e.g. [3,4]) as the "motion profile" is not as "constant" as on a car driving around.
 
Calibration: If you have only a single sensor and the poses are anyway in the coordinate system of the sensor, you have to simply have a identity matrix in the calibration file. (You find some issues here that discuss this)
 
The point_labeler currently works just with the point clouds in KITTI format; therefore you would have to produce the .bin files (or write something to directly ready the rosbags by modifying the code)
 
Motion compensation aka point cloud deskewing: In KITTI, the point clouds are already provided in way that accounts for the motion of the sensor while scanning the environment; i.e., the movement between the start of a turn of the spinning LIDAR is already accounted for. KISS-ICP does a point cloud deskewing internally to estimate the poses, but it would be good to also produce the point clouds for labeling such that points are deskewed.
  
However, all the steps above require probably some fine-tuning with the specific platform that you are using, as I expect that the drone movement is more "high frequent" than a "normal" car drive; therefore, integrating information of an IMU is beneficial to account for the motion of the sensor while scanning the environment (it's similar to the rolling shutter vs. a global shutter with images).
 
References
 [1] https://github.com/PRBonn/kiss-icp
 [2] https://github.com/PRBonn/kiss-slam
 [3] https://github.com/YibinWu/LIO-EKF
 [4] https://github.com/hku-mars/FAST_LIO

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  jbehley left a comment (jbehley/point_labeler#95)
 
As I didn't produce the KITTI dataset itself, I can only say what is needed to potentially label the point clouds with the tool (which was built for working with KITTI format files.) There are potentially other ways, but it's your decision to use it or not.
  
Pose estimation: The point_labeler uses the poses provided in the poses.txt. These can be estimated with something like KISS-ICP or KISS-SLAM that provide an easy-to-use off-the-shelf solution that is easy to setup: pip install is all you need. It usually provide good poses out of the box without extensive hyperparameter tuning. For drone data, it might be even worth considering a LiDAR-intertial SLAM or odmetry approach (e.g. [3,4]) as the "motion profile" is not as "constant" as on a car driving around.
 
Calibration: If you have only a single sensor and the poses are anyway in the coordinate system of the sensor, you have to simply have a identity matrix in the calibration file. (You find some issues here that discuss this)
 
The point_labeler currently works just with the point clouds in KITTI format; therefore you would have to produce the .bin files (or write something to directly ready the rosbags by modifying the code)
 
Motion compensation aka point cloud deskewing: In KITTI, the point clouds are already provided in way that accounts for the motion of the sensor while scanning the environment; i.e., the movement between the start of a turn of the spinning LIDAR is already accounted for. KISS-ICP does a point cloud deskewing internally to estimate the poses, but it would be good to also produce the point clouds for labeling such that points are deskewed.
  
However, all the steps above require probably some fine-tuning with the specific platform that you are using, as I expect that the drone movement is more "high frequent" than a "normal" car drive; therefore, integrating information of an IMU is beneficial to account for the motion of the sensor while scanning the environment (it's similar to the rolling shutter vs. a global shutter with images).
 
References
 [1] https://github.com/PRBonn/kiss-icp
 [2] https://github.com/PRBonn/kiss-slam
 [3] https://github.com/YibinWu/LIO-EKF
 [4] https://github.com/hku-mars/FAST_LIO
